File: Programming Practice/Python/template.py
#!/usr/bin/python

# Imports
import sys, os

# Global variables
exec_name = os.path.basename(__file__)
exec_path = os.path.realpath(__file__)
exec_dir = os.path.dirname(os.path.realpath(__file__))
original_cwd = os.getcwd()

# Main
class Main:
    @staticmethod
    def main():
        # Define method variables
        exit_code = 0
        
        # Do code
        sys.stdout.write("Executable name: " + exec_name + "\n")
        sys.stdout.write("Executable path: " + exec_path + "\n")
        sys.stdout.write("Executable dir: " + exec_dir + "\n")
        sys.stdout.write("Original cwd: " + original_cwd + "\n")
        
        # Done
        return exit_code

# Execute
if __name__ == "__main__":
    sys.exit(Main.main())

File: coding_practice/Perl/template.pl
#!/usr/bin/perl

# Main Class
package Main;

# Imports
use strict;
use warnings;
use FindBin;
use Cwd;

# Global variables
my $script_name = "$FindBin::RealScript";
my $script_path = "$FindBin::RealBin/$FindBin::RealScript";
my $script_dir = "$FindBin::RealBin";
my $original_cwd = getcwd();

# Main
sub main {
    # Define subroutine variables
    my $exit_code = 0;
    
    # Do code
    print(STDOUT "Script name: $script_name\n");
    print(STDOUT "Script path: $script_path\n");
    print(STDOUT "Script dir: $script_dir\n");
    print(STDOUT "Original cwd: $original_cwd\n");
    
    # Done
    return $exit_code;
}

# Execute
exit(Main::main());

# End Main Class
1;

